openhands/sdk/llm/utils/metrics.py

```diff
@@ -109,6 +110,10 @@ class Metrics(MetricsSnapshot):
         default_factory=list, description="List of token usage records"
     )
 
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._lock = threading.Lock()
+
     @field_validator("accumulated_cost")
```

Comment:
@openhands we should not write an __init__ like this for pydantic basemodels 

please look at how we set private attributes in `ConversationStats` and and follow a similar pattern here